File: hs/cs50-main/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h> // use for strlen
#include <math.h>   // use for round

int count_sentences(string text)
{
    // similar code to count_letters
    int total = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            total++;
        }
    }
    return total;
}

int count_words(string text)
{
    // similar code to count_letters
    int total = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            total++;
        }
    }
    return total;
}

int count_letters(string text)
{
    int total = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] < 91 && text[i] > 64) || (text[i] < 123 && text[i] > 96))
        {
            total++;
        }
    }
    return total;
}

int main(void)
{
    string h = get_string("text: ");
    // i had to use floats otherwise the answer would just stick to 1.84 because s is always 0
    float W = count_words(h);
    float L = (count_letters(h) / W) * 100;
    float S = (count_sentences(h) / W) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8); // round needed for int 
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: hs/cs50-main/sentimental-readability/readability.py
from cs50 import get_string


# took code from my c assignment and translated it to python
def count_letters(txt):
    total = 0
    for i in range(len(txt)):
        value = ord(txt[i])
        if ((value < 91 and value > 64) or (value < 123 and value > 96)):
            total += 1
    return total


def count_words(txt):
    total = 1
    for i in range(len(txt)):
        if (txt[i] == ' '):
            total += 1
    return total


def count_sentences(txt):
    total = 0
    for i in range(len(txt)):
        value = ord(txt[i])
        if (value == 46 or value == 33 or value == 63):
            total += 1
    return total


text = get_string("text: ")
W = count_words(text)
L = (count_letters(text) / W) * 100 
S = (count_sentences(text) / W) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)
if (index < 1):
    print("Before Grade 1")
elif (index >= 16):
    print("Grade 16+")
else:
    print(f"Grade {index}")